### ISSUE TITLE
Why can't we evaluate wer/cer while training using PEFT?

### ISSUE THREAD

**Issue by MightyStud** [huggingface/peft#810]
### System Info

In the tutorial notebook for Whisper fine tuning using PEFT it mentions that we can't use `compute_metric` in training arguments. And the reason is as stated in the [notebook](https://github.com/huggingface/peft/blob/main/examples/int8_training/peft_bnb_whisper_large_v2_training.ipynb) :

> INT8 training required autocasting. predict_with_generate can't be passed to Trainer because it internally calls transformer's generate without autocasting leading to errors.
> 
>  compute_metrics shouldn't be passed to Seq2SeqTrainer as seen below. (commented out)

I'd like to autocast it manually or fix this issue in one way or another. 
computing WER/CER during training is important to validate the model output in a continuous fashion since train/val loss is somewhat meaning less in this kind of model/tasks.

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

run the notebook

### Expected behavior

being able to compute wer/cer while training.

**Comment by MightyStud** [comment]
can't we pass the conversion to `preprocess_logits_for_metrics` in `Seq2SeqTrainer` ?

**Comment by Tony2055866** [comment]
以下来自chatgpt：
是否提供 compute_metrics 对于优化模型的过程来说是没有直接的影响的。compute_metrics 主要用于模型性能的评估和监控，而不是在模型的参数更新中直接使用。它在训练或验证期间被用来计算一些性能指标，以便更全面地了解模型的表现。

在训练过程中，最关键的是损失函数（Loss Function）。优化的目标是通过最小化损失函数，使得模型的预测结果尽可能接近真实标签。损失函数直接影响参数的更新，从而影响模型的训练。

compute_metrics 在训练期间提供了一种评估模型性能的机制，但它通常不会影响模型的参数更新。它更多地用于监控模型在训练集或验证集上的表现，为用户提供更详细的反馈。

总体来说，compute_metrics 是一种用于评估模型性能的工具，而损失函数是训练期间优化模型的主要工具。提供 compute_metrics 对于更全面地了解模型在任务上的表现是有帮助的，但不会直接影响模型的参数更新。
